templates/trust/contracts/TrustTemplate.sol

```diff
@@ -77,182 +83,182 @@ contract TrustTemplate is BaseTemplate {
-        require(_hasDaoCache(msg.sender) && _hasAppsCache(msg.sender), ERROR_MISSING_CACHE_TOKENS_FOR_SENDER);
-        require(multiSigKeys.length == MULTI_SIG_EXTERNAL_KEYS_AMOUNT, ERROR_BAD_MULTI_SIG_KEYS_LENGTH);
+    function setupMultiSig(address[] _multiSigKeys) public returns (MultiSigWallet) {
+        require(_hasDaoCache(msg.sender) && _hasAppsCache(msg.sender), ERROR_MISSING_SENDER_CACHE);
+        require(_multiSigKeys.length == MULTI_SIG_EXTERNAL_KEYS_AMOUNT, ERROR_BAD_MULTI_SIG_KEYS_LENGTH);
 
         Kernel dao = _getDaoCache(msg.sender);
-        MultiSigWallet multiSig = _setupMultiSig(dao, multiSigKeys);
+        MultiSigWallet multiSig = _setupMultiSig(dao, _multiSigKeys);
         emit DeployMultiSig(address(multiSig));
         return multiSig;
     }
 
-    function _setupApps(Kernel dao, address[] beneficiaryKeys, address[] heirs, uint256[] heirsStake, uint256 blockedHeirsSupply) internal {
-        ACL acl = ACL(dao.acl());
-        Agent agent = _installNonDefaultAgentApp(dao);
-        Vault vault = _installVaultApp(dao);
-        Finance finance = _installFinanceApp(dao, vault, 30 days);
-        (Voting holdVoting, Voting heirsVoting, TokenManager holdTokenManager, TokenManager heirsTokenManager) = _installTokenApps(dao);
+    function _setupApps(Kernel _dao, address[] _beneficiaryKeys, address[] _heirs, uint256[] _heirsStake, uint256 _blockedHeirsSupply) internal {
+        ACL acl = ACL(_dao.acl());
+        Agent agent = _installNonDefaultAgentApp(_dao);
+        Vault vault = _installVaultApp(_dao);
+        Finance finance = _installFinanceApp(_dao, vault, 30 days);
+        (Voting holdVoting, Voting heirsVoting, TokenManager holdTokenManager, TokenManager heirsTokenManager) = _installTokenApps(_dao);
 
-        _mintHoldTokens(acl, holdTokenManager, beneficiaryKeys);
-        _mintHeirsTokens(acl, heirsTokenManager, heirs, heirsStake, blockedHeirsSupply);
+        _mintHoldTokens(acl, holdTokenManager, _beneficiaryKeys);
+        _mintHeirsTokens(acl, heirsTokenManager, _heirs, _heirsStake, _blockedHeirsSupply);
 
         _createVaultPermissions(acl, vault, finance, holdVoting);
         _createFinancePermissions(acl, finance, holdVoting, holdVoting);
-        _createTokenManagerPermissions(acl, holdTokenManager, holdVoting);
-        _createTokenManagerPermissions(acl, heirsTokenManager, heirsVoting);
         _createEvmScriptsRegistryPermissions(acl, holdVoting, holdVoting);
-        _createVotingPermissions(acl, holdTokenManager, holdVoting);
-        _createVotingPermissions(acl, heirsTokenManager, heirsVoting);
-        _createAgentPermission(acl, agent, agent.EXECUTE_ROLE(), holdVoting, heirsVoting);
-        _createAgentPermission(acl, agent, agent.RUN_SCRIPT_ROLE(), holdVoting, heirsVoting);
+        _createCustomVotingPermissions(acl, holdTokenManager, holdVoting);
+        _createCustomAgentPermissions(acl, agent, holdVoting, heirsVoting);
```

Comment:
It's a bit odd the Agent permissions are done in-between the Voting permission assignments